libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -64,20 +74,19 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
       LOG(log_dg_) << "Received PBFT vote " << vote->getHash() << " with PBFT block " << pbft_block->getBlockHash();
     } else {
       vote = std::make_shared<Vote>(packet_data.rlp_[i]);
+      vote_hash = vote->getHash();
+
+      // Synchronization point in case multiple threads are processing the same vote at the same time
+      if (!seen_votes_.insert(vote_hash)) {
+        LOG(log_tr_) << "Received vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
+                     << ") already seen.";
+        continue;
+      }
       pbft_block = nullptr;
       peer_chain_size = 0;
       LOG(log_dg_) << "Received PBFT vote " << vote->getHash();
     }
 
-    const auto vote_hash = vote->getHash();
-
-    // Synchronization point in case multiple threads are processing the same vote at the same time
-    if (!seen_votes_.insert(vote_hash)) {
-      LOG(log_dg_) << "Received vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
-                   << ") already seen.";
-      continue;
-    }
```

Comment:
why was this moved from here ?

Reply:
I noticed that we have a lot of these repeating logs. If we are connected to 20 nodes, we have 20 of these for each received vote:
Taraxa_N1_09192022_144827_06792.log:0x00007fd1b37fe640 10c405dc… PBFT_VOTE_PH [2022-09-19 14:48:53.200986] DEBUG: Received PBFT vote #1e53029a… with PBFT block #65eef53e…
Taraxa_N1_09192022_144827_06792.log:0x00007fd1b37fe640 10c405dc… PBFT_VOTE_PH [2022-09-19 14:48:53.201078] DEBUG: Received vote #1e53029a… (from d8c2b208…) already seen.
Taraxa_N1_09192022_144827_06792.log:0x00007fd1c9ffb640 10c405dc… PBFT_VOTE_PH [2022-09-19 14:48:53.265493] DEBUG: Received PBFT vote #1e53029a… with PBFT block #65eef53e…
Taraxa_N1_09192022_144827_06792.log:0x00007fd1c9ffb640 10c405dc… PBFT_VOTE_PH [2022-09-19 14:48:53.265692] DEBUG: Received vote #1e53029a… (from 58d0c916…) already seen.
Taraxa_N1_09192022_144827_06792.log:0x00007fd1c9ffb640 10c405dc… PBFT_VOTE_PH [2022-09-19 14:48:53.269122] DEBUG: Received PBFT vote #1e53029a… with PBFT block #65eef53e…
.........................

Reply:
Besides the logging if we receive a vote we already know, I think it is better not to construct a pbft block. Now we construct same pbft block 20 times